From 95b1ccc06398d93a54cca215d3e5ff42bd0b6f56 Mon Sep 17 00:00:00 2001
From: Duanqiang Wen <duanqiangwen@net-swift.com>
Date: Fri, 4 Nov 2022 09:39:47 +0800
Subject: [PATCH] openeuler: config: add NGBE MODODULE CONFIG
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 95b1ccc06398d93a54cca215d3e5ff42bd0b6f56
Modified-by-SEL: Yes, the change in openeuler_defconfig are dropped


driver inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I61PSD
CVE: NA

1.add CONFIG_NETSWIFT_NGBE=m in x64/arm64 openeuler_defconfig
2.add config for ngbe module in netswift Kconfig

Signed-off-by: Duanqiang Wen <duanqiangwen@net-swift.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 drivers/net/ethernet/netswift/Kconfig |   49 ++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)

--- a/drivers/net/ethernet/netswift/Kconfig
+++ b/drivers/net/ethernet/netswift/Kconfig
@@ -61,4 +61,53 @@ config TXGBE_SYSFS
 	  Say Y if you want to setup sysfs for these devices.
 
 	  If unsure, say N.
+
+config NGBE
+	tristate "Netswift PCI-Express Gigabit Ethernet support"
+	depends on PCI
+	imply PTP_1588_CLOCK
+	help
+	  This driver supports Netswift gigabit ethernet adapters.
+	  For more information on how to identify your adapter, go
+	  to <http://www.net-swift.com>
+
+	  To compile this driver as a module, choose M here. The module
+	  will be called ngbe.
+
+config NGBE_HWMON
+	bool "Netswift PCI-Express Gigabit adapters HWMON support"
+	default n
+	depends on NGBE && HWMON && !(NGBE=y && HWMON=m)
+	help
+	  Say Y if you want to expose thermal sensor data on these devices.
+
+	  If unsure, say N.
+
+config NGBE_DEBUG_FS
+	bool "Netswift PCI-Express Gigabit adapters debugfs support"
+	default n
+	depends on NGBE
+	help
+	  Say Y if you want to setup debugfs for these devices.
+
+	  If unsure, say N.
+
+config NGBE_POLL_LINK_STATUS
+	bool "Netswift PCI-Express Gigabit adapters poll mode support"
+	default n
+	depends on NGBE
+	help
+	  Say Y if you want to turn these devices to poll mode instead of interrupt-trigged TX/RX.
+
+	  If unsure, say N.
+
+config NGBE_SYSFS
+	bool "Netswift PCI-Express Gigabit adapters sysfs support"
+	default n
+	depends on NGBE
+	help
+	  Say Y if you want to setup sysfs for these devices.
+
+	  If unsure, say N.
+
 endif # NET_VENDOR_NETSWIFT
